llvm/lib/Transforms/Utils/GlobalStatus.cpp

```diff
@@ -172,9 +172,20 @@ static bool analyzeGlobalAux(const Value *V, GlobalStatus &GS,
           return true;
         GS.StoredType = GlobalStatus::Stored;
       } else if (const auto *CB = dyn_cast<CallBase>(I)) {
-        if (!CB->isCallee(&U))
-          return true;
-        GS.IsLoaded = true;
+        bool KnownIntrinsic = false;
+        if (const IntrinsicInst *II = dyn_cast<IntrinsicInst>(I)) {
+          if (II->getIntrinsicID() == Intrinsic::threadlocal_address &&
+              II->getArgOperand(0) == V) {
```

Comment:
Can `II->getArgOperand(0) == V` actually fail here?  We know the intrinsic is a use of `V`, and it only has one argument.

Reply:
I was contemplating whether there could be operand bundles. But that probably isn't really a thing for this intrinsic, removing the check.